bin/ch-dir2squash

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+set -e
+
+libexec="$(cd "$(dirname "$0")" && pwd)"
+. "${libexec}/base.sh"
+
+# shellcheck disable=SC2034
+usage=$(cat <<EOF
+Pack a filesystem directory into a squashfs.
+
+Usage:
+
+  $ $(basename "$0") INDIR OUTDIR [ARGS ...]
+EOF
+)
+
+sentinel=WEIRD_AL_YANKOVIC
+
+parse_basic_args "$@"
+
+if [ $# -lt 2 ]; then
+    usage
+fi
+
+indir=$1
+outdir=$2
+
+shift 2
+
+# Check input directory exists
+if [ ! -e "$indir" ]; then
+    echo "can't squash: ${indir} does not exist" 1>&2
+    exit 1
+fi
+# Check input directory is a directory
+if [ ! -d "$indir" ]; then
+    echo "can't squash: ${indir} is not a directory" 1>&2
+    exit 1
+fi
+# Check input directory is a charliecloud image
+if [ ! -e "${indir}/${sentinel}" ]; then
+    echo "${indir} is NOT a Charliecloud iamge" 1>&2
+    exit 1
+fi
+
+# Check OUTDIR exists and is a directory
+if [ ! -e "$outdir" ]; then
+    echo "can't squash: ${outdir} does not exist" 1>&2
+    exit 1
+fi
+if [ ! -d "$outdir" ]; then
+    echo "can't squash: ${outdir} is not a directory" 1>&2
+    exit 1
+fi
+
```

Comment:
There is a lot of repetitive exits and stdout redirection. I'd prefer to make use of a `fatal` function and use short circuiting for checking. For example:

```bash
fatal () {
    printf '%s\n\n' "$1" 1>&2
    exit 1
}

[ -e "$indir" ] || fatal "can't squash: $indir does not exits"
[ -d "$indir" ] || fatal "can't squash: $indir is not a directory"
```

Reply:
I think that would be more concise and readable. I think that would probably be something for a separate PR to create the function and update all the helper scripts to make use of it.